x-pack/plugins/alerting/server/authorization/alerting_authorization_kuery.ts

```diff
@@ -25,42 +32,35 @@ export interface AlertingAuthorizationFilterOpts {
 interface AlertingAuthorizationFilterFieldNames {
   ruleTypeId: string;
   consumer: string;
+  spaceIds?: string;
 }
 
-const esQueryConfig: EsQueryConfig = {
-  allowLeadingWildcards: true,
-  dateFormatTZ: 'Zulu',
-  ignoreFilterIfFieldNotInIndex: false,
-  queryStringOptions: { analyze_wildcard: true },
-};
-
 export function asFiltersByRuleTypeAndConsumer(
   ruleTypes: Set<RegistryAlertTypeWithAuth>,
-  opts: AlertingAuthorizationFilterOpts
-): KueryNode | JsonObject {
-  const kueryNode = nodeBuilder.or(
-    Array.from(ruleTypes).reduce<KueryNode[]>((filters, { id, authorizedConsumers }) => {
-      ensureFieldIsSafeForQuery('ruleTypeId', id);
-      filters.push(
-        nodeBuilder.and([
-          nodeBuilder.is(opts.fieldNames.ruleTypeId, id),
-          nodeBuilder.or(
-            Object.keys(authorizedConsumers).map((consumer) => {
-              ensureFieldIsSafeForQuery('consumer', consumer);
-              return nodeBuilder.is(opts.fieldNames.consumer, consumer);
-            })
-          ),
-        ])
-      );
-      return filters;
-    }, [])
-  );
-
+  opts: AlertingAuthorizationFilterOpts,
+  alertSpaceId?: string
+): KueryNode | BooleanFilter {
   if (opts.type === AlertingAuthorizationFilterType.ESDSL) {
-    return toElasticsearchQuery(kueryNode, undefined, esQueryConfig);
```

Comment:
What do you think of incorporating spaceId (if defined) into the kueryNode as well and continuing to use `toElasticsearchQuery` to generate the ES DSL query? That way we can ensure the find authorization filter is the same regardless of query language. I'm worried it will be to easy to diverge the queries if we're building them independently.

Reply:
Yea, that's a great point. I had thought that we needed it to be a terms query which I couldn't get using the kueryNode, but it seems to be working as expected with a simple match. I've updated the code per your suggestions.